OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs

```diff
@@ -71,32 +72,37 @@ void INotifyCreated.Created(Actor self)
 			experience = self.TraitOrDefault<PlayerExperience>();
 		}
 
-		void UpdateEarnedThisMinute()
-		{
-			EarnedSamples.Add(EarnedThisMinute);
-			earnedAtBeginningOfMinute = resources != null ? resources.Earned : 0;
-		}
-
-		void UpdateArmyThisMinute()
-		{
-			ArmySamples.Add(ArmyValue);
-		}
-
 		void ITick.Tick(Actor self)
 		{
-			if (self.Owner.WinState != WinState.Undefined)
-				return;
-
 			ticks++;
 
 			var timestep = self.World.IsReplay ? replayTimestep : self.World.Timestep;
-
-			if (ticks * timestep >= 60000)
+			if (ticks * timestep >= 30000)
 			{
 				ticks = 0;
-				UpdateEarnedThisMinute();
-				UpdateArmyThisMinute();
+
+				if (ArmyValue != 0 || self.Owner.WinState == WinState.Undefined)
```

Comment:
Not sure its worth fixing, but things will go wrong if we stop updating (self.Owner.WinState -> lost) and then at some time later `ArmyValue` becomes non-zero. If we want to protect against this we could keep guard bools to make sure we never reenable this if this becomes false.

Reply:
That would still leave the graph in bogus state (no value display although the value went up again). The only way to properly fix this I see is adding 0 values to the graph.